src/Libraries/Microsoft.Extensions.Telemetry/Logging/Logger.cs

```diff
@@ -3,179 +3,80 @@
     {
-        if (!IsEnabled(logLevel))
+        _nextLogger = nextLogger;
+        _enrichers = enrichers;
+        _includeStackTraces = includeStackTraces;
+        _maxStackTraceLength = maxStackTraceLength;
+        _redactorProvider = redactorProvider;
+
+        var bag = new PropertyBag();
+        foreach (var enricher in staticEnrichers)
         {
-            return;
+            enricher.Enrich(bag);
         }
 
-        LogMethodHelper propertyBag;
-        LogMethodHelper? rentedHelper = null;
-
-        try
-        {
-            if (state is LogMethodHelper helper && _provider.CanUsePropertyBagPool)
-            {
-                propertyBag = helper;
-            }
-            else
-            {
-                rentedHelper = GetHelper();
-                propertyBag = rentedHelper;
-
-                switch (state)
-                {
-                    case IReadOnlyList<KeyValuePair<string, object?>> stateList:
-                        rentedHelper.AddRange(stateList);
-                        break;
-
-                    case IEnumerable<KeyValuePair<string, object?>> stateList:
-                        rentedHelper.AddRange(stateList);
-                        break;
-
-                    case null:
-                        break;
-
-                    default:
-                        rentedHelper.Add("{OriginalFormat}", state);
-                        break;
-                }
-            }
-
-            foreach (var enricher in _provider.Enrichers)
-            {
-                enricher.Enrich(propertyBag);
-            }
-
-            if (exception != null && _provider.IncludeStackTrace)
-            {
-                propertyBag.Add(ExceptionStackTrace, GetExceptionStackTrace(exception, _provider.MaxStackTraceLength));
-            }
+        _staticProperties = bag.ToArray();
+    }
 
-            var record = CreateLogRecord(
-                _provider.IncludeScopes ? ScopeProvider : null,
-                TimeProvider.GetUtcNow().UtcDateTime,
-                _categoryName,
-                logLevel,
-                eventId,
-                _provider.UseFormattedMessage ? formatter(state, exception) : null,
+    public IDisposable? BeginScope<TState>(TState state)
+        where TState : notnull
+    {
+        return _nextLogger.BeginScope(state);
+    }
 
-                // This parameter needs to be null for OpenTelemetry.Exporter.Geneva to pick up LogRecord.StateValues (the last parameter).
-                // This is equivalent to using OpenTelemetryLogger with ParseStateValues option set to true.
-                null,
-                exception,
-                propertyBag);
+    public bool IsEnabled(LogLevel logLevel)
+    {
+        // TODO: is this the best we can do here? Should/could the enabled state be tracked in this instance so we can avoid the interface call?
+        return _nextLogger.IsEnabled(logLevel);
```

Comment:
@davidfowl @noahfalk Is this the best I can do here? I'd love to avoid the overhead of this interface call.

Similarly, the check at the top of the ModernPath/R9Path/LegacyPath methods are also inducing these interface calls.